src/server.c

```diff
@@ -2942,7 +2942,10 @@ void adjustOpenFilesLimit(void) {
 
                 /* We failed to set file limit to 'bestlimit'. Try with a
                  * smaller limit decrementing by a few FDs per iteration. */
-                if (bestlimit < decr_step) break;
+                if (bestlimit < decr_step) {
```

Comment:
@yoav-steinberg doesn't it need to be:
```suggestion
                if (bestlimit <= decr_step) {
```
i.e. we never want the while condition to break on it's own

Reply:
@oranagra , I think your suggested change is fine.

Reply:
@garenchan thank you.
do you agree with my other statements?
1. this whole fix is only relevant if for some reason the system limit is extremely low (lower than some 32).
2. the change i suggested for `<` vs `<=` may not be important, since considering the we'll exit the loop with bestlimit of 0, it'll be lower than oldlimit, and we'll revert it anyway after the loop.

Reply:
@oranagra , Yes, I agree with your other statements.